docs/variants/qemu_q35/development.md

```diff
@@ -0,0 +1,138 @@
+# Dasharo (UEFI) v0.1.0 for QEMU Q35
+
+## Intro
+
+1. Please check your qemu setup and launch qemu outside
+docker instance (host machine) using built OVMF firmware image.
+
++ QEMU Installation:
+    - [Install QEMU on your linux distro](<https://www.qemu.org/download/#linux>)
+
+ ```bash
+ qemu-system-x86_64 -drive if=pflash,format=raw,file=Build/OvmfX64/DEBUG_GCC5/FV/OVMF.fd -debugcon file:debug.log -global isa-debugcon.iobase=0x402 -global ICH9-LPC.disable_s3=1 -net none -machine q35,smm=on
+ ```
+
++ `-drive` indicate device is pflash with firmware image of built OVMF.fd image.
+
++ `-debugcon file:debug.log -global isa-debugcon.iobase=0x402`
+default OVMF build writes debug messages to IO port **0x402**.
+The following qemu command line options save them in the file called debug.log.
+
++ `-global ICH9-LPC.disable_s3=1` SMM is put to use in the S3 suspend
+and resume infrastructure, and in the UEFI variable driver stack.
+Similarly, a pflash-backed variable store is a requirement.
+
++ `-net none` disables the OptionROM execution which
+will interfere with the firmware image, if not disabled.
+
++ `-machine q35,smm=on` For SMM to work,
+only Q35 machines are supported hence the machine type.
+
+1. After executing the above qemu command,
+   qemu boots into UEFI built-in shell and to BIOS selection area.
+
+1. The features which are enabled in `OvmfPkgX64.dsc` can be verified at `BIOS menu`.
+   At `Device Manager section` Dasharo features can
+   be verified in `Dasharo System Features` section.
+
+## Useful Tips for modifying the changes in .DSC & .FDF
+
+### Follow below steps, to rebuild the firmware image and experiments with features
+
++ If any code is added to C file or any library functions are invoked in other files,
+rebuild the OVMF image with the following command.
+
+```bash
+build -D DEBUG_ON_SERIAL_PORT -D BOOTLOADER=COREBOOT -a IA32 -a X64 -t GCC5 -b DEBUG -p OvmfPkg/OvmfPkgX64.dsc
+```
+
+1. By making some changes in the `OvmfPkgX64.dsc` file, one can add features like
+OPAL, SATA security support for QEMU Q35 machine and also for specific platforms.
+
+1. An example of where the changes
+can be made in the `OvmfPkgX64.dsc` file is given below.
+
+> The following changes describe about defining in the **Defines Section** of `OvmfPkgX64.dsc`
+
+```bash
+# Defines for default states.  These can be changed on the command line.
+# -D FLAG=VALUE
+#
+(...)
+DEFINE SATA_PASSWORD_ENABLE    = TRUE
+DEFINE OPAL_PASSWORD_ENABLE    = TRUE
+```
+
+> Include the appropriate libraries in the **Libraries Section** of `OvmfPkgX64.dsc`.
+
+```bash
+#
+# OPAL_PASSWORD
+#
+
+!if $(OPAL_PASSWORD_ENABLE) == TRUE
+TcgStorageCoreLib|SecurityPkg/Library/TcgStorageCoreLib/TcgStorageCoreLib.inf
+TcgStorageOpalLib|SecurityPkg/Library/TcgStorageOpalLib/TcgStorageOpalLib.inf
+!endif
+S3BootScriptLib|MdePkg/Library/BaseS3BootScriptLibNull/BaseS3BootScriptLibNull.inf
```

Comment:
Please align that with my review from edk2 PR. NULL S3BootScriptLib is redundant here I believe.